### PR TITLE
PCT-1510: fix for explain

### PR DESCRIPTION
**Pull request by arvenil**
https://jira.percona.com/browse/PCT-1510

### CODE REVIEW

**Comment by cfsalguero** [comment]
:+1:

**Comment by mpaolino** [comment]
:+1:

File: mysql/mysql.go [hunk 1]
```diff
@@ -202,6 +205,31 @@ func (c *Connection) GetGlobalVarNumber(varName string) float64 {
...
 }
 
+func (c *Connection) AtLeastVersion(v string) (bool, error) {
+	mysqlVersion := c.GetGlobalVarString("version") // Version in the form m.n.o-ubuntu
+	return AtLeastVersion(mysqlVersion, v)
+}
+
+// Check if version v2 is equal or higher than v1 (v2 >= v1)
+// v2 can be in form m.n.o-ubuntu
+func AtLeastVersion(v1, v2 string) (bool, error) {
```
Comment (daniel-nichter) [inline mysql/mysql.go]: Is there another way to write this? Having two funcs with the same name confused me after first. I saw this one with two args, but then saw it called below with 1 arg, but then called in test with 2 args. This seems like an anti-pattern.

Comment (arvenil) [inline mysql/mysql.go]: > Is there another way to write this?

Exactly. Nothing better come to my mind. Better idea?

Comment (daniel-nichter) [inline mysql/mysql.go]: Yes... `(c *Connection) AtLeastVersion()` is not needed. Almost any func with only 1 or 2 lines is not needed, i.e. is unnecessary wrapper stuff, unless there's some specific design reason for it (e.g. in one of my PRs you'll see I've done it several times, but the reason: I'm maintaining what I think will become an interface, so the wrapper funcs are facades/adapters for this future interface). This func (the 1 arg version) is the important one with the logic, and that logic is generic, so move it out to `pct/` (which has become a dumping ground of sorts for such things), then in the caller, `jsonExplain()`, do:
```
pct.AtLeastVeresion(c.GetGlobalVarString("version"), "5.6.5")
```
Then `(c *Connection) AtLeastVersion()` can be removed.

Comment (daniel-nichter) [inline mysql/mysql.go]: I thought @cfsalguero did a func like `AtLeastVersion()` already for the installer? If so, then we really need this generic logic in a shared place.

**Pull request merged**